### ISSUE TITLE
[E::sam_parse1] query name too long

### ISSUE THREAD

**Issue by williamrowell** [lh3/minimap2#858]
I'm attempting to align one haplotype of a hifiasm assembly (v0.15) to reference.  This worked fine for haplotype 1, but there's a `query name too long` error for haplotype 2.  The FASTA files were generated by `gfatools gfa2fasta` (v0.4), followed by compression with bgzip.

```text
> minimap2 -t 20 -L --secondary=no --eqx -ax asm5 -R '@RG\tID:NA19240_hap2\tSM:NA19240' reference/human_GRCh38_no_alt_analysis_set.fasta <(samples/NA19240/hifiasm/NA19240.asm.bp.hap2.p_ctg.fasta.gz) | samtools sort -@ 3 -m 8G > samples/NA19240/hifiasm/NA19240.hap2.GRCh38.bam

[M::mm_idx_gen::51.372*1.47] collected minimizers
[M::mm_idx_gen::59.833*2.23] sorted minimizers
[M::main::59.833*2.23] loaded/built the index for 195 target sequence(s)
[M::mm_mapopt_update::68.990*2.06] mid_occ = 136
[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 195
[M::mm_idx_stat::73.779*1.99] distinct minimizers: 214664461 (92.23% are singletons); average occurrences: 1.372; average spacing: 10.529; total length: 3099922541
[E::sam_parse1] query name too long
```

I started with minimap2 v2.17, and this actually worked fine, but I hit a SegFault attempting to align another sample so I decided to update to the most recent minimap2.

```text
> minimap2 --version
2.24-r1122
```

The query names are all 12 characters long.  I've tried uncompressing the input first.

```text
> zgrep '>' samples/NA19240/hifiasm/NA19240.asm.bp.hap2.p_ctg.fasta.gz | awk '{print length($1)}' | datamash min 1 max 1 mean 1 median 1
12      12      12      12
```

**Comment by lh3** [comment]
```
<(samples/NA19240/hifiasm/NA19240.asm.bp.hap2.p_ctg.fasta.gz)
```

Remove `<()`.

**Comment by lh3** [comment]
Then you can generate SAM first and then see if there are long query names. The error report is from samtools.

**Comment by williamrowell** [comment]
Thanks.  Generated intermediate SAM file and query names were still only 11 characters.  Bug in samtools 1.10, fixed by 1.12.
